src/metabase/sync/sync_metadata/fields/sync_metadata.clj

```diff
@@ -79,11 +79,17 @@
                       new-database-type)
            {:database_type new-database-type})
          (when new-base-type?
-           (log/infof "Base type of %s has changed from ''%s'' to ''%s''."
+           (log/infof "Base type of %s has changed from ''%s'' to ''%s''. This field will be refingerprinted and analyzed."
                      (common/field-metadata-name-for-logging table metabase-field)
                      old-base-type
                      new-base-type)
-           {:base_type new-base-type})
+           {:base_type           new-base-type
+            :effective_type      new-base-type
+            :coercion_strategy   nil
+            ;; reset fingerprint version so this field will get re-fingerprinted and analyzed
+            :fingerprint_version 0
+            :fingerprint         nil
+            :semantic_type       nil})
```

Comment:
It's not clear why `:semantic_type` must be nil here. It took me a while to realise it was because of this:
https://github.com/metabase/metabase/blob/224b904a0ca5892134d83eda715a519eb27aa159/src/metabase/sync/analyze/fingerprint/fingerprinters.clj#L261

I think you should either document this here, or better, create a function that "resets" fingerprints and analysis that lives in an appropritate namespace under `metabase.sync.analyze`